File: plugin/src/main/java/me/kingtux/versionfile/VFMain.java
package me.kingtux.versionfile;


import org.gradle.api.Plugin;
import org.gradle.api.Project;

public class VFMain implements Plugin<Project> {

    @Override
    public void apply(Project target) {
        VFTask vfTask = new VFTask(target);
        target.task("vftask").doLast(task -> {
            vfTask.createFile(target);
        });
    }
}

File: plugin/src/main/kotlin/me.kingtux.versionfile/VFMain.kt
package me.kingtux.versionfile

import org.gradle.api.Plugin
import org.gradle.api.Project
import org.gradle.api.Task

class VFMain : Plugin<Project> {
    override fun apply(target: Project) {
        val vfTask = VFTask(target)
        target.task("vftask").doLast { vfTask.createFile(target) }
    }
}